### ISSUE TITLE
Panic in cubic solver: alpha-near-zero assertion fires during 3-set ellipse fit

### ISSUE THREAD

**Issue by jolars** [jolars/eunoia#32]
## Summary

For a benign 3-set ellipse fit, the cubic solver in `crates/eunoia/src/math/polynomial.rs:52` panics with:

```
thread '<unnamed>' panicked at .../eunoia-0.2.0/src/math/polynomial.rs:52:5:
alpha must be non-zero for a cubic equation (got -0.000000000000007305103077216678)
```

The fitter then reports:
```
InvalidCombination("Optimization failed: all optimization restarts failed (panicked or errored)")
```

Surfaced while migrating eulerr's backend to eunoia: this is the only input in eulerr's `tests/testthat/test-reproducibility.R` that fails the new pipeline. The `alpha` value (`-7.3e-15`) is effectively zero — the solver should treat that as the quadratic case, not panic.

## Reproduction (eunoia 0.2.0)

```rust
use eunoia::{geometry::shapes::Ellipse, DiagramSpecBuilder, Fitter, InputType};

let spec = DiagramSpecBuilder::new()
    .set("a", 3491.0)
    .set("b", 3409.0)
    .set("c", 3503.0)
    .intersection(&["a","b"], 120.0)
    .intersection(&["a","c"], 114.0)
    .intersection(&["b","c"], 132.0)
    .intersection(&["a","b","c"], 126.0)
    .input_type(InputType::Exclusive)
    .build()
    .unwrap();

// Panics on at least one optimization restart; with the SA fallback
// enabled (default), every restart panics and `.fit()` returns
// `Err(InvalidCombination("Optimization failed: all optimization restarts failed..."))`.
let layout = Fitter::<Ellipse>::new(&spec).seed(1).fit();
```

Triggered with `loss = LossType::SumSquared` (default) and `loss = LossType::MaxSquared`.
Circle fits for the same spec succeed.

## Likely cause

`solve_cubic` (or whatever consumer calls it) passes `alpha ≈ 0` because of floating-point cancellation during conic intersection / segment-area evaluation. The assertion `alpha != 0` should be a tolerance check (e.g. `|alpha| < ε → fall back to the quadratic branch`) rather than a hard panic.

## Suggested fix

In `crates/eunoia/src/math/polynomial.rs:52`:

- Replace the panic with a tolerance check: when `alpha.abs() < CUBIC_DEGENERATE_EPS` (e.g. `1e-12`), defer to the quadratic solver.
- Alternatively, return a `Result` so the caller can decide — but a tolerance fallback is the standard treatment for degenerate cubics.

## Context

- eulerr migration: backend switch from Rcpp/C++ to extendr/eunoia.
- Only failure of this kind across eulerr's full reproducibility test grid (17 specs × 2 shapes × 3 losses × 2 aggregators).
